multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -270,6 +284,55 @@ func (c *StaleResCleanupController) cleanupClusterInfoResourceExport(commonArea
 	return nil
 }
 
+func (c *StaleResCleanupController) cleanupMemberClusterAnnounce() error {
+	clusterSetList := &mcsv1alpha1.ClusterSetList{}
+	if err := c.List(ctx, clusterSetList, &client.ListOptions{}); err != nil {
+		return err
+	}
+	memberClusterAnnounceList := &mcsv1alpha1.MemberClusterAnnounceList{}
+	if err := c.List(ctx, memberClusterAnnounceList, &client.ListOptions{}); err != nil {
+		return err
+	}
+
+	for _, memberClusterAnnounce := range memberClusterAnnounceList.Items {
+		var exist bool
+		var needToDelete bool
+		if len(clusterSetList.Items) != 0 {
+			members := clusterSetList.Items[0].Spec.Members
+			for _, member := range members {
+				if member.ClusterID == memberClusterAnnounce.ClusterID {
+					exist = true
+					break
+				}
+			}
+			if !exist {
+				klog.InfoS("Cleaning up stale MemberClusterAnnounce. It does exist in the member cluster list", "MemberClusterAnnounce", klog.KObj(&memberClusterAnnounce))
+				needToDelete = true
+			} else {
+				lastUpdateTime, err := time.Parse(time.RFC3339, memberClusterAnnounce.Annotations[commonarea.TimestampAnnotationKey])
+				if err == nil && time.Now().Sub(lastUpdateTime) > 5*time.Minute {
+					klog.InfoS("Cleaning up stale MemberClusterAnnounce. It has not been updated for 5 minutes", "MemberClusterAnnounce", klog.KObj(&memberClusterAnnounce))
+					needToDelete = true
+				} else if err != nil {
+					klog.ErrorS(err, "Cleaning up stale MemberClusterAnnounce. Failed to parse last update time", "MemberClusterAnnounce", klog.KObj(&memberClusterAnnounce))
+					needToDelete = true
+				}
```

Comment:
I think you you can replace this with below one:
```go
if err == nil && time.Now().Sub(lastUpdateTime) < 5*time.Minute {
  needToDelete =false
} else {
  ...log error ...
  needToDelete = true
}
```

Reply:
Done,.